### ISSUE TITLE
Unable to import pool after system update

### ISSUE THREAD

**Issue by ACiDGRiM** [openzfs/zfs#10486]
# System information
Distribution Name	|  Centos
Distribution Version	| 8.2
Linux Kernel	|  4.18.0-193.6.3.el8_2.centos.plus.x86_64
Architecture	| x86_64
ZFS Version	| 0.8.4-1
SPL Version	| 0.8.4-1

### Describe the problem you're observing
zpool suspends and locks up zfs module when importing pool
Able to import module readonly=on with no issues

I've downgraded to zfs 0.8.3 and booted to the last kernel this is working on, but the issue still occurs

### Describe how to reproduce the problem
zpool import tank

here is the pool properties from the pool when imported read only:
```
# zpool get all
NAME                 PROPERTY                       VALUE                          SOURCE
zfs_storage-array01  size                           72.8T                          -
zfs_storage-array01  capacity                       48%                            -
zfs_storage-array01  altroot                        -                              default
zfs_storage-array01  health                         ONLINE                         -
zfs_storage-array01  guid                           4921511236475250392            -
zfs_storage-array01  version                        -                              default
zfs_storage-array01  bootfs                         -                              default
zfs_storage-array01  delegation                     on                             default
zfs_storage-array01  autoreplace                    off                            default
zfs_storage-array01  cachefile                      none                           local
zfs_storage-array01  failmode                       wait                           default
zfs_storage-array01  listsnapshots                  off                            default
zfs_storage-array01  autoexpand                     off                            default
zfs_storage-array01  dedupditto                     0                              default
zfs_storage-array01  dedupratio                     1.04x                          -
zfs_storage-array01  free                           37.6T                          -
zfs_storage-array01  allocated                      35.2T                          -
zfs_storage-array01  readonly                       on                             -
zfs_storage-array01  ashift                         12                             local
zfs_storage-array01  comment                        -                              default
zfs_storage-array01  expandsize                     -                              -
zfs_storage-array01  freeing                        0                              -
zfs_storage-array01  fragmentation                  0%                             -
zfs_storage-array01  leaked                         0                              -
zfs_storage-array01  multihost                      on                             local
zfs_storage-array01  checkpoint                     -                              -
zfs_storage-array01  load_guid                      12886774754137858919           -
zfs_storage-array01  autotrim                       on                             local
zfs_storage-array01  feature@async_destroy          enabled                        local
zfs_storage-array01  feature@empty_bpobj            active                         local
zfs_storage-array01  feature@lz4_compress           active                         local
zfs_storage-array01  feature@multi_vdev_crash_dump  enabled                        local
zfs_storage-array01  feature@spacemap_histogram     active                         local
zfs_storage-array01  feature@enabled_txg            active                         local
zfs_storage-array01  feature@hole_birth             active                         local
zfs_storage-array01  feature@extensible_dataset     active                         local
zfs_storage-array01  feature@embedded_data          active                         local
zfs_storage-array01  feature@bookmarks              enabled                        local
zfs_storage-array01  feature@filesystem_limits      enabled                        local
zfs_storage-array01  feature@large_blocks           enabled                        local
zfs_storage-array01  feature@large_dnode            enabled                        local
zfs_storage-array01  feature@sha512                 enabled                        local
zfs_storage-array01  feature@skein                  enabled                        local
zfs_storage-array01  feature@edonr                  active                         local
zfs_storage-array01  feature@userobj_accounting     active                         local
zfs_storage-array01  feature@encryption             active                         local
zfs_storage-array01  feature@project_quota          active                         local
zfs_storage-array01  feature@device_removal         enabled                        local
zfs_storage-array01  feature@obsolete_counts        enabled                        local
zfs_storage-array01  feature@zpool_checkpoint       enabled                        local
zfs_storage-array01  feature@spacemap_v2            active                         local
zfs_storage-array01  feature@allocation_classes     enabled                        local
zfs_storage-array01  feature@resilver_defer         enabled                        local
zfs_storage-array01  feature@bookmark_v2            enabled                        local
```

### Include any warning/errors/backtraces from the system logs
Here are logs in dmesg during import without readonly=on

```

[ 7539.820680] WARNING: MMP writes to pool 'zfs_storage-array01' have not succeeded in over 166002 ms; suspending pool. Hrtime 7539820794974
[ 7539.820683] WARNING: Pool 'zfs_storage-array01' has encountered an uncorrectable I/O failure and has been suspended.
```

**Comment by fryfrog** [comment]
Does it still show up in `zpool list` and `zpool status`?

**Comment by ACiDGRiM** [comment]
No, when I run any zpool commands they hang and I have to re-connect to the ssh session. I can see the previously run zpool applications in ps aux.

I have removed as many variables as I can, including connecting only one path to the SAS enclosure and disabling multipathd on boot.

This occurs on both systems in a pacemaker cluster, I'm attempting to reinstall one of them with Centos 8.1, however it will be great if I can identify why this is failing. Thank you for the response.

**Comment by ofaaland** [comment]
Hi @ACiDGRiM , I believe you are encountering the issue fixed by this patch:
https://github.com/openzfs/zfs/pull/10912

That patch is included in both zfs-0.8.5 and zfs-2.0.0 releases, so upgrading to either should get you going again.

If you're unable to upgrade, one workaround is to set ```zfs_multihost_fail_intervals = 0``` before you import the pool, which tells ZFS not to suspend the pool due to MMP delays or failures.

**Comment by ACiDGRiM** [comment]
Thanks for the update even after all this time!
I will attempt the update when I can, (un) fortunately this pool remains in this same state.

**Comment by ofaaland** [comment]
Sorry you didn't get an earlier response.  Try the workaround I mentioned.  Note that the bug is at import time.  So if you wanted to, you could set that module parameter to 0 before the import, and then set it back to your normal setting after the import.  The pacemaker resource agent that imports the pool is probably a shell script that would be easy to modify.

**Comment by ACiDGRiM** [comment]
Thank you, this issue is confirmed resolved for me in v0.8.5.
I had tried the zfs_multihost_fail_intervals parameter with no success in my case.
